sycl/unittests/CMakeLists.txt

```diff
@@ -56,3 +56,9 @@ if (NOT WIN32)
   add_subdirectory(xpti_trace)
 endif()
 
+# Library unit testing
+include(AddSYCLLibraryUnitTest)
+add_custom_target(check-sycl-unittests-libs)
+add_dependencies(check-sycl-unittests check-sycl-unittests-libs)
+
+add_subdirectory(syclcompat)
```

Comment:
We had a discussion and it was decided that it was best to have the headers, docs and tests in the relevant directories, rather than all together. For the tests, we were recommended to submit those within the existing `sycl/unittests` directory. But if you think it would be better to have them separated, we could also do that.

Reply:
+ @steffenlarsen , @sergey-semenov

Reply:
Pinging @steffenlarsen and @sergey-semenov one more time :)

Reply:
I think the current unittest location makes more sense considering the location of syclcompat headers and docs. Unless you're suggesting to move everything to a dedicated syclcompat directory, in which case, I don't really have a strong preference one way or the other.